ci-operator/step-registry/upi/deprovision/azure/upi-deprovision-azure-commands.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+set +e
+echo "Deprovisioning cluster ..."
+if [[ -f ${SHARED_DIR}/metadata.json ]]; then
+    INFRA_ID="$(jq -r .infraID ${SHARED_DIR}/metadata.json)"
+    AZURE_AUTH_CLIENT_ID=$(cat ${SHARED_DIR}/AZURE_AUTH_CLIENT_ID)
+    AZURE_AUTH_CLIENT_SECRET=$(cat ${SHARED_DIR}/AZURE_AUTH_CLIENT_SECRET)
+    AZURE_AUTH_TENANT_ID=$(cat ${SHARED_DIR}/AZURE_AUTH_TENANT_ID)
+    az login --service-principal -u $AZURE_AUTH_CLIENT_ID -p "$AZURE_AUTH_CLIENT_SECRET" --tenant $AZURE_AUTH_TENANT_ID
+    az group delete --name ${INFRA_ID}-rg --yes || true
```

Comment:
We should probably check that the resource group exists before deleting it. That way we don't have to swallow any real errors that may occur when trying to delete an existing resource group.